sim-administration/sim-batch-management/es2plus/es2plus.go

```
@@ -0,0 +1,401 @@
+	if r.Method == "POST" {
+		r.ParseForm()
+		request = append(request, "\n")
+		request = append(request, r.Form.Encode())
+	}
+	// Return the request as a string
+	return strings.Join(request, "\n")
+}
+
+func newUuid() (string, error) {
+    uuid, err := uuid.NewRandom()
+    if err != nil {
+    		return "", err
+    	}
+    return uuid.URN(), nil
+}
+
+func newEs2plusHeader(client *Es2PlusClient) (*ES2PlusHeader) {
+   functionCallIdentifier, err := newUuid()
+   if err != nil  {
+        panic(err)
+   }
+   return &ES2PlusHeader{FunctionCallIdentifier: functionCallIdentifier, FunctionRequesterIdentifier: client.requesterId}
+}
+
+
+func marshalUnmarshalGenericEs2plusCommand(client *Es2PlusClient, es2plusCommand string,  payload interface{}, result interface{}) error {
+
+    // Serialize payload as json.
+	jsonStrB, err := json.Marshal(payload)
+	if err != nil {
+		return  err
+	}
+
+    if client.printPayload {
+    	fmt.Print("Payload ->", string(jsonStrB))
+    }
+
+    // Get the result of the HTTP POST as a byte array
+    // that can be deserialized into json.  Fail fast
+    // an error has been detected.
+	responseBytes, err := executeGenericEs2plusCommand(jsonStrB, client.hostport, es2plusCommand,  client.httpClient, client.printHeaders)
+	if err != nil {
+		return  err
+	}
+
+    // Return error code from deserialisation, result is put into
+    // result via referenced object.
+	return json.Unmarshal(responseBytes, result)
+}
+
+
+func executeGenericEs2plusCommand(jsonStrB []byte, hostport string, es2plusCommand string, httpClient *http.Client, printHeaders bool) ([]byte, error) {
+
+    // Build and execute an ES2+ protocol request by using the serialised JSON in the
+    // byte array jsonStrB in a POST request.   Set up the required
+    // headers for ES2+ and content type.
+	url := fmt.Sprintf("https://%s/gsma/rsp2/es2plus/%s", hostport, es2plusCommand)
+	req, _ := http.NewRequest("POST", url, bytes.NewBuffer(jsonStrB))
```

Don't ignore the error.